github-core/src/main/java/com/itachallenge/githubcore/service/GithubApiService.java

```
@@ -0,0 +1,40 @@
+package com.itachallenge.githubcore.service;
+
+import com.itachallenge.githubcore.exception.GithubApiException;
+import com.itachallenge.githubcore.exception.GithubUserNotFoundException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.web.reactive.function.client.WebClient;
+import reactor.core.publisher.Mono;
+
+
+public class GithubApiService implements IGithubApiService{
+
+    private static final Logger log = LoggerFactory.getLogger(GithubApiService.class);
+    private final WebClient webClient;
+
+    public GithubApiService(WebClient.Builder builder, String baseUrl) {
+        this.webClient = builder.baseUrl(baseUrl).build();
+    }
+
+    @Override
+    public Mono<Boolean> userExists(String username) {
+        return webClient.get()
+                .uri("/users/{username}", username)
+                .retrieve()
+                .onStatus(HttpStatusCode::is4xxClientError, r -> {
+                    log.info("GitHub user not found: {}", username);
+                    return Mono.error(new GithubUserNotFoundException(username));
+                })
+                .onStatus(HttpStatusCode::is5xxServerError, r -> {
```

Here, you do need to throw an exception because a GitHub 5xx (or timeout) is a real technical error.
The only thing I suggest is not using a generic name like GithubApiException, but something more precise:

GithubUnavailableException → for 5xx/timeout (external service down/not responding).

---

Yes, I have changed it to GithubUnavailableException,